### PR TITLE
logrotate: Add ACL support

### PR DESCRIPTION
**Pull request by neheb**
Various Makefile cleanups.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

Maintainer: @bk138

### CODE REVIEW

**Comment by bk138** [comment]
This might do it :-)

**Comment by cotequeiroz** [comment]
Shouldn't we make this optional?  I'd keep it `default n`, unless the use case is more common than I envision.

**Comment by bk138** [comment]
> Shouldn't we make this optional? I'd keep it `default n`, unless the use case is more common than I envision.

I for one have no strong opinion on that.

**Comment by neheb** [comment]
User in the linked thread looks like he does not build his packages. Either the package is split up into two variants or ACL just gets enabled. My vote is for the latter.

**Comment by cotequeiroz** [comment]
AFAICT, kernel ACL support is [disabled by default](https://github.com/openwrt/openwrt/blob/c3e31b6a9b040912be0b532f9bf29d91c6c74382/config/Config-kernel.in#L731).

**Comment by neheb** [comment]
Interesting... I think the user is using logrotate under TurrisOS, which has an interntionally bloated kernel. Probably has ACL enabled too. It seems I have logrotate installed on my Omnia too...

File: utils/logrotate/Makefile [hunk 1]
```diff
@@ -9,22 +9,24 @@ include $(TOPDIR)/rules.mk
...
+PKG_LICENSE:=GPL-2.0-or-later
+PKG_LICENSE_FILES:=COPYING
 
 include $(INCLUDE_DIR)/package.mk
 
 define Package/logrotate
   SECTION:=utils
   CATEGORY:=Utilities
-  DEPENDS:=+libpopt
+  DEPENDS:=+libpopt +USE_FS_ACL_ATTR:libacl
```
Comment (cotequeiroz) [inline utils/logrotate/Makefile]: I was thinking about doing it like rsync, with an explicit option, and `MENU:=1`.  It makes it easier to find than at the Kernel Config.  My reasoning is, apart from someone with kernel support not wanting to enable it for logrotate is that you may enable ACL in a fs by fs basis. Besides when a user needs this, the easiest/first thing to check would be the package, but I recon that could well be just baseless speculation on my part :smile:.  You may add a reference to the kernel options at the package help, so user can enable it in the kernel and actually use ACLs.  Anyway, you should add whatever symbol you're using to `PKG_CONFIG_DEPENDS`.

**Comment by n8v8R** [comment]
@cotequeiroz 
> AFAICT, kernel ACL support is disabled by default.

https://git.openwrt.org
config/Config-kernel.in	
 729         config USE_FS_ACL_ATTR
 741                 default y if USE_FS_ACL_ATTR
 746                 default y if USE_FS_ACL_ATTR
 751                 default y if USE_FS_ACL_ATTR
 771                 default y if USE_FS_ACL_ATTR
 776                 default y if USE_FS_ACL_ATTR
 781                 default y if USE_FS_ACL_ATTR
 785                 default y if USE_FS_ACL_ATTR
 802                 default y if USE_FS_ACL_ATTR
 807                 default y if USE_FS_ACL_ATTR
 812                 default y if USE_FS_ACL_ATTR
___

@neheb 
> I think the user is using logrotate under TurrisOS, which has an interntionally bloated kernel. 

Not sure I get this - `logrotate` is bloatware?
____
The repo is offering various ACL related userland. There is little point with ACL support enabled in the kernel that userland is being curtailed however.

**Comment by neheb** [comment]
Not what I meant. TurrisOS does not have space and RAM limitations like most devices running OpenWrt.

**Comment by n8v8R** [comment]
What is the cost in disk space and RAM to have ACL support enabled in the kernel and `logrotate` too?

ACL deployment in the kernel is inconsistent, for some targets it is enabled for others not.

File: utils/logrotate/Makefile (hunk 1 above)
Comment (neheb) [reply utils/logrotate/Makefile]: I think this is what you meant.

Comment (cotequeiroz) [reply utils/logrotate/Makefile]: Exactly. :smiley:

**Comment by n8v8R** [comment]
and thank y'all for the effort!

**Pull request merged**